CRCTray.cs

```diff
@@ -237,28 +237,37 @@ async private void StartMenu_Click(object sender, EventArgs e)
             TaskHelpers.TryTask(Tasks.SendTelemetry, Actions.ClickStart).NoAwait();
 
             // Check using get-config if pullSecret is configured
-            var pullsecret = await TaskHelpers.TryTask(Tasks.GetPullSecret);
-            if (!pullsecret)
+            try
             {
-                var pullSecretForm = new PullSecretForm();
-                pullSecretForm.ShowDialog();
-                var pullSecretContent = pullSecretForm.PullSecret;
-                if (pullSecretContent == String.Empty)
-                {
-                    TrayIcon.NotifyWarn(@"No Pull Secret was provided. Cannot start cluster without pull secret.");
-                    return;
-                }
-
-                TaskHelpers.TryTask(Tasks.SendTelemetry, Actions.EnterPullSecret).NoAwait();
 
-                // Store pullsecret; returns false is not able to
-                var stored = await TaskHelpers.TryTask(Tasks.SetPullSecret, pullSecretContent);
-                if(!stored)
+                var pullsecret = await TaskHelpers.TryTaskWithRetry(Tasks.GetPullSecret);
+                if (!pullsecret)
                 {
-                    TrayIcon.NotifyError(@"Pull-secret not stored. Please check if valid format is used.");
-                    return;
-                }
+                    var pullSecretForm = new PullSecretForm();
+                    pullSecretForm.ShowDialog();
+                    var pullSecretContent = pullSecretForm.PullSecret;
+                    if (pullSecretContent == String.Empty)
+                    {
+                        TrayIcon.NotifyWarn(@"No Pull Secret was provided. Cannot start cluster without pull secret.");
+                        return;
+                    }
+
+                    TaskHelpers.TryTask(Tasks.SendTelemetry, Actions.EnterPullSecret).NoAwait();
+
+                    // Store pullsecret; returns false is not able to
+                    var stored = await TaskHelpers.TryTask(Tasks.SetPullSecret, pullSecretContent);

```

Comment:
Wouldn't this be impacted by taskcanceled too?

Reply:
Should we use `TryTaskWithRetry` for all the the calls as all of them has a chance of getting cancelled..